ethexe/contracts/src/ScaleCodec.sol

```diff
@@ -0,0 +1,508 @@
+        bytes memory result = new bytes(1);
+        result[0] = value;
+        return result;
+    }
+
+    function bytes2Tobytes(bytes2 value) public pure returns (bytes memory) {
+        bytes memory result = new bytes(2);
+        result[0] = value[0];
+        result[1] = value[1];
+        return result;
+    }
+
+    function bytes4Tobytes(bytes4 value) public pure returns (bytes memory) {
+        bytes memory result = new bytes(4);
+        for (uint256 i = 0; i < 4; i++) {
+            result[i] = value[i];
+        }
+        return result;
+    }
+
+    function bytes8Tobytes(bytes8 value) public pure returns (bytes memory) {
+        bytes memory result = new bytes(8);
+        for (uint256 i = 0; i < 8; i++) {
+            result[i] = value[i];
+        }
+        return result;
+    }
+
+    function bytes16Tobytes(bytes16 value) public pure returns (bytes memory) {
+        bytes memory result = new bytes(16);
+        for (uint256 i = 0; i < 16; i++) {
+            result[i] = value[i];
+        }
+        return result;
+    }
+
+    function bytes32Tobytes(bytes32 value) public pure returns (bytes memory) {
+        bytes memory result = new bytes(32);
+        for (uint256 i = 0; i < 32; i++) {
+            result[i] = value[i];
+        }
+        return result;
+    }
+
+    function bytesToBytes32(bytes memory value, uint256 offset) public pure returns (bytes32 result) {
+        assembly {
+            result := mload(add(add(value, 0x20), offset))
+        }
+    }
+
+    function insertBytes20To(bytes20 data, bytes memory destination, uint256 offset) internal pure {
+        assembly {
+            mstore(add(add(destination, 0x20), offset), data)
+        }
+    }
+
+    function insertBytes32To(bytes32 data, bytes memory destination, uint256 offset) internal pure {
+        assembly {
+            mstore(add(add(destination, 0x20), offset), data)
+        }
+    }
+
+    function insertBytesTo(bytes memory data, bytes memory destination, uint256 offset) internal pure {
+        assembly {
+            let data_len := mload(data)
+            let dest_ptr := add(add(destination, 0x20), offset)
+            let data_ptr := add(data, 0x20)
+            for { let i := 0 } lt(i, data_len) { i := add(i, 1) } {
+                let v := mload(add(data_ptr, i))
+                mstore(add(dest_ptr, i), v)
+            }
+        }
+    }
+
+    function encodeBool(bool value) public pure returns (bytes memory) {
+        bytes memory result = new bytes(1);
+        encodeBoolTo(value, result, 0);
+        return result;
+    }
+
+    function encodeBoolTo(bool value, bytes memory destination, uint256 offset) internal pure {
+        if (value) {
+            destination[offset] = 0x01;
+        } else {
+            destination[offset] = 0x00;
+        }
+    }
+
+    function decodeBool(bytes memory _bytes, uint256 offset) public pure returns (bool) {
+        return _bytes[offset] == 0x01;
```

Comment:
what if its 0x02?

Reply:
you should return result here maybe
for optimised code impl decodeBoolUnchecked

Reply:
similar everywhere

Reply:
Since there is no native Result type in Solidity I think it's not the best idea.
Maybe calling `revert` is better if the bytes are wrong?